### PR TITLE
Throw exceptions instead of returning them

### PR DESCRIPTION
**Pull request by nidhijaju**
As a follow up to https://github.com/w3c/webauthn/pull/1706#discussion_r828811241, this PR aims to address the necessary changes on the Credential Management side to throw exceptions in all of the Credential internal methods: [`[[CollectFromCredentialStore]]`](https://w3c.github.io/webappsec-credential-management/#algorithm-collect-creds), [`[[DiscoverFromExternalSource]]`](https://w3c.github.io/webappsec-credential-management/#algorithm-discover-creds), [`[[Create]]`](https://w3c.github.io/webappsec-credential-management/#algorithm-create-cred), and [`[[Store]]`](https://w3c.github.io/webappsec-credential-management/#algorithm-store-cred). 

All of the credential types have also been updated in the places:
- Web Authentication: https://github.com/w3c/webauthn/pull/1706
- WebOTP: https://github.com/WICG/web-otp/pull/57
- Password Credentials: this PR
- Federated Credentials: this PR

@nsatragno: Would you be able to take a look at this? Thank you!

Fixes https://github.com/w3c/webappsec-credential-management/issues/197

***
<a href="https://pr-preview.s3.amazonaws.com/nidhijaju/webappsec-credential-management/pull/196.html" title="Last updated on Apr 1, 2022, 12:16 AM UTC (1441cab)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webappsec-credential-management/196/dbabb7b...nidhijaju:1441cab.html" title="Last updated on Apr 1, 2022, 12:16 AM UTC (1441cab)">Diff</a>

### CODE REVIEW

**Review by nsatragno — COMMENTED** [review]
This overall looks good, thanks!

Two overall pieces of feedback:
1. We should also throw (instead of returning) exceptions for `[[Store]]` and `[[CollectFromCredentialStore]]` for consistency. Otherwise we'll run into the same issue when another spec extends those methods. (and sorry -- I should have mentioned this earlier)
2. Can you file an issue for this PR and link it?

Thank you so much for your work here Nidhi!

**Comment by nidhijaju** [comment]
> * We should also throw (instead of returning) exceptions for `[[Store]]` and `[[CollectFromCredentialStore]]` for consistency. Otherwise we'll run into the same issue when another spec extends those methods. (and sorry -- I should have mentioned this earlier)

Done :)

> * Can you file an issue for this PR and link it?

Here is the issue: https://github.com/w3c/webappsec-credential-management/issues/197 and I've linked to it in the original description as well.

**Comment by nidhijaju** [comment]
FYI, I think the remaining bikeshed error should be fixed by https://github.com/w3c/webappsec-credential-management/pull/193

**Review by nsatragno — COMMENTED** [review]
Thank you so much! This is essentially ready to land now. A few minor comments:

File: index.bs
Comment (nidhijaju) [reply index.bs:1120]: Same as above.

**Review by nsatragno — APPROVED** [review]
LGTM

Will merge once #193 gets merged. Thank you so much!

**Comment by nidhijaju** [comment]
Okay, I've rebased on main now :)

Thank you so much for reviewing!

**Pull request merged**
